sdk/src/signer.rs

```diff
@@ -76,10 +76,29 @@ use async_trait::async_trait;
 #[async_trait]
 pub trait AsyncSigner: Sync {
     /// Returns a new byte array which is a signature over the original.
-    async fn sign(&self, data: &[u8]) -> Result<Vec<u8>>;
+    async fn sign(&self, data: Vec<u8>) -> Result<Vec<u8>>;
```

Comment:
Why does this need to consume `data`?

If that _does_ need to remain in place, this PR should be marked with the `(MINOR)` tag in the title since it's an API breaking change.

Reply:
This was not addressed.